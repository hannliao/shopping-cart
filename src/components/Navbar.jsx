import '../styles/Navbar.css';
import { Link } from 'react-router-dom';

const Navbar = () => {
  return (
    <>
      <div className="banner">
        <p>free shipping on all orders $150+</p>
      </div>
      <nav>
        <div className="header">
          <Link to="/">
            <h1>basic basics</h1>
          </Link>

          <ul className="icons">
            <li>
              <button>
                <img src="/search-outline.svg" alt="" />
              </button>
            </li>
            <li>
              <button>
                <img src="/person-outline.svg" alt="" />
              </button>
            </li>
            <li>
              <button>
                <img src="/cart-outline.svg" alt="" />
              </button>
            </li>
          </ul>
        </div>

        <ul className="shop">
          <li>
            <Link to="/collections/all-products">Shop All</Link>
          </li>
          <li>
            <Link to="/collections/women">Women</Link>
          </li>
          <li>
            <Link to="/collections/men">Men</Link>
          </li>
          <li>
            <Link to="/collections/accessories">Accessories</Link>
          </li>
          <li>
            <Link to="/collections/shoes">Shoes</Link>
          </li>
        </ul>
      </nav>
    </>
  );
};

export default Navbar;
