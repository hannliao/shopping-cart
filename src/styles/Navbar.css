.banner {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--banner);
  font-size: 0.8em;
  padding: 0.6em;
}

h1 {
  font-family: 'fugaz';
  font-size: 2.5em;
}

nav {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 50px;
}

.header {
  display: flex;
  justify-content: center;
  align-items: center;
  padding-top: 1em;
}

.shop {
  padding: 0.4em;
  width: 100%;
  display: flex;
  justify-content: center;
  border-top: 0.7px solid var(--border);
  border-bottom: 0.7px solid var(--border);
  font-size: 0.8em;
}

.shop a:hover {
  text-decoration: underline;
  text-underline-offset: 5px;
}

nav ul {
  display: flex;
}

nav li {
  padding: 0.2em 1.1em;
}

.icons {
  position: absolute;
  right: 50px;
}

nav img {
  width: 25px;
}

.cart-count {
  font-size: 12px;
  font-weight: 500;
  width: 20px;
  position: relative;
  left: 12px;
  bottom: 38px;
  border-radius: 25px;
  background-color: darkred;
  color: white;
}
