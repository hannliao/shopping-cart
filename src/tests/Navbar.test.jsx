import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('navbar', () => {
  it('renders navbar', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('logo links to home page', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(screen.getByRole('link', { name: /basic basics/i })).toHaveAttribute(
      'href',
      '/'
    );
  });

  it('shop all links to all products page', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(screen.getByRole('link', { name: /shop all/i })).toHaveAttribute(
      'href',
      '/collections/all-products'
    );
  });

  it('women links to womens clothing page', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(screen.getByRole('link', { name: /women/i })).toHaveAttribute(
      'href',
      '/collections/women'
    );
  });

  it('men links to mens clothing page', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(screen.getByRole('link', { name: /Men/ })).toHaveAttribute(
      'href',
      '/collections/men'
    );
  });

  it('accessories links to accessories page', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(screen.getByRole('link', { name: /accessories/i })).toHaveAttribute(
      'href',
      '/collections/accessories'
    );
  });

  it('shoes links to shoes page', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(screen.getByRole('link', { name: /shoes/i })).toHaveAttribute(
      'href',
      '/collections/shoes'
    );
  });
});
