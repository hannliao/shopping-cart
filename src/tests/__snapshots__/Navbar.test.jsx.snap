// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`navbar > renders navbar 1`] = `
<div>
  <header>
    <div
      class="banner"
    >
      <p>
        free shipping on all orders $150+
      </p>
    </div>
    <nav>
      <div
        class="header"
      >
        <a
          href="/"
        >
          <h1>
            basic basics
          </h1>
        </a>
        <ul
          class="icons"
        >
          <li>
            <a>
              <img
                alt=""
                src="/search-outline.svg"
              />
            </a>
          </li>
          <li>
            <a>
              <img
                alt=""
                src="/person-outline.svg"
              />
            </a>
          </li>
          <li>
            <a
              href="/cart"
            >
              <img
                alt=""
                src="/cart-outline.svg"
              />
            </a>
          </li>
        </ul>
      </div>
      <ul
        class="shop"
      >
        <li>
          <a
            href="/collections/all-products"
          >
            Shop All
          </a>
        </li>
        <li>
          <a
            href="/collections/women"
          >
            Women
          </a>
        </li>
        <li>
          <a
            href="/collections/men"
          >
            Men
          </a>
        </li>
        <li>
          <a
            href="/collections/accessories"
          >
            Accessories
          </a>
        </li>
        <li>
          <a
            href="/collections/shoes"
          >
            Shoes
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
